### PR TITLE
Create Homework_2,py

### PR DESCRIPTION
**Pull request by Verantina007**

### CODE REVIEW

File: Homework_2,py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+
+import random #import library for useing random function
+import string #import library for working with strings
+
+n = random.randint (2, 10) # length of list with dictionaries
+
+res_list=[] # empty list for adding disctionaries
+
+for i in range(n): # circle for the number of elements in list
```
Comment (romkabauer) [inline Homework_2,py]: If you have time, try to use list comprehension. Not obligatory to fix

File: Homework_2,py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+# if key is only in one dict - take it as is,
+# example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+
+# checking res_list=[{'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}]
+
+final_dict = {} # final dictionary for adding the result
+
+for char in string.ascii_lowercase: # go through all letters in the alphabet
+    temp_values={} # temporary dictionary for each letter with the number of dictionary and value for this key
+    for i in range(n): # circle through all dictionaries in the list
```
Comment (romkabauer) [inline Homework_2,py]: Why not iterate through res_list? enumerate() will help on this

File: Homework_2,py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        if res_list[i].get(char) is not None: # exclude None values - letters that do not exist in dictionary
+            temp_values[char+'_'+str(i+1)]=res_list[i].get(char) # add number of dictionary for the letter and value for this key
```
Comment (romkabauer) [inline Homework_2,py]: We don't need to add index for the values, which are unique. Please update your condition for if

File: Homework_2,py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+# example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+
+# checking res_list=[{'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}]
+
+final_dict = {} # final dictionary for adding the result
+
+for char in string.ascii_lowercase: # go through all letters in the alphabet
+    temp_values={} # temporary dictionary for each letter with the number of dictionary and value for this key
+    for i in range(n): # circle through all dictionaries in the list
+        if res_list[i].get(char) is not None: # exclude None values - letters that do not exist in dictionary
```
Comment (romkabauer) [inline Homework_2,py]: You can just leave `if res_list[i].get(char):`. The result will be the same.

**Review by romkabauer — CHANGES_REQUESTED** [review]

File: Homework_2,py (hunk 1 above)
Comment (Verantina007) [reply Homework_2,py]: Done

File: Homework_2,py (hunk 2 above)
Comment (Verantina007) [reply Homework_2,py]: Done

File: Homework_2,py (hunk 3 above)
Comment (Verantina007) [reply Homework_2,py]: Done

File: Homework_2,py (hunk 4 above)
Comment (Verantina007) [reply Homework_2,py]: Done

File: Homework_2,py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+# dict's values should be a number (0-100),
+# example: [{'a': 5, 'b': 7, 'g': 11}, {'a': 3, 'c': 35, 'g': 42}]
+
+import random #import library for useing random function
+import string #import library for working with strings
+
+# generate a list of dictionaries
+# n - number of dictionaries (list elements)
+# m - number of elements in each dictionary
+res_list = [({random.choice(string.ascii_lowercase): random.randint(0, 100) for m in range(random.randint (1, 26))}) for n in range(random.randint(2, 10))]
```
Comment (romkabauer) [inline Homework_2,py:14]: Perfect:)

File: Homework_2,py [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+for char in string.ascii_lowercase: # go through all letters in the alphabet
+    temp_values={} # temporary dictionary for each letter with the number of dictionary and value for this key
+    for i in enumerate(res_list): # circle through all dictionaries in the list
+        if i[1].get(char):
+            temp_values[char + '_' + str(i[0] + 1)] = i[1].get(char) # add number of dictionary for the letter and value for this key
+    for key, value in temp_values.items():  # circle through temporary dictionary for each letter
+        if len(temp_values)==0: # if letter doesn't exist in any dictionary
+            break
+        if len(temp_values)==1: # if letter exists only in one dictionary
```
Comment (romkabauer) [inline Homework_2,py]: here is the logic bound to one object, let's use `elif` statement to show that the logic is interrelated

File: Homework_2,py [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+# if key is only in one dict - take it as is,
+# example: {'a_1': 5, 'b': 7, 'c': 35, 'g_2': 42}
+
+# checking res_list=[{'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}, {'a':1, 'b':2, 'c':3}]
+
+final_dict = {} # final dictionary for adding the result
+
+for char in string.ascii_lowercase: # go through all letters in the alphabet
+    temp_values={} # temporary dictionary for each letter with the number of dictionary and value for this key
+    for i in enumerate(res_list): # circle through all dictionaries in the list
```
Comment (romkabauer) [inline Homework_2,py]: you can use enumerate in a more efficient way like `for ind, next_dict in enumerate(res_list):` so you can easily access index and object of iteration inside the cycle

File: Homework_2,py (hunk 6 above)
Comment (Verantina007) [reply Homework_2,py]: Done

File: Homework_2,py (hunk 7 above)
Comment (Verantina007) [reply Homework_2,py]: Done

**Review by romkabauer — APPROVED** [review]

**Pull request merged**